src/routes/notification/index.ts

```diff
@@ -249,13 +251,51 @@ router.post('/send-email/:configId', async (req, res) => {
         for (const obj of tempRecords) {
           const data = obj?.data;
           for (const [key, value] of Object.entries(data)) {
-            if (mongoose.isValidObjectId(value) && typeof value === 'string') {
+            const userField = fields.find((x) => {
+              return x.type === 'users' && x.name === key;
+            });
+            const ownerField = fields.find((x) => {
+              return x.type === 'owner' && x.name === key;
+            });
+            if (mongoose.isValidObjectId(value) && typeof value == 'string') {
               const project = mergeArrayOfObjects(nestedFields[key]) ?? {};
               Object.assign(project, { _id: 0 });
               const record = await Record.findById(value, project);
               data[key] = record;
+              console.log('HIT');
+            }
+            if (value instanceof Array) {
+              if (userField) {
+                const options = await getUsersOptions(userField.applications);
+                const users = value.map((user: any) => {
+                  const values = options.filter((opt) => {
+                    return opt.value.toString() === user;
+                  });
+                  return _.map(values, 'text');
+                });
+
+                const usersJoined = users.join(', ');
+                data[key] =
+                  usersJoined.length > 0 ? usersJoined : value.join(', ');
+              }
+              if (ownerField) {
+                const ownerOptions = await getOwnerOptions(
```

Comment:
@HasnatL-Adapptlimited 
you need to put that somewhere else
let's say you have n records and each has owner field being filled, then you do n times the same query